nemo_skills/evaluation/metrics/if_metrics.py

```diff
@@ -21,35 +21,11 @@ class IFMetrics(BaseMetrics):
     # loosely adapted from
     # https://github.com/google-research/google-research/blob/master/instruction_following_eval/evaluation_main.py
 
-    required_keys = ['follow_instruction_list', 'instruction_id_list']
-
-    def __init__(self):
-        self.reset()
-
-    def _update_single_stat(self, stats_dict, elems):
-        """Will update using the pass@k strategy (just pass a single-element list to get greedy)."""
-        # has to be the same across all elements as they are solutions for the same question
-        instruction_id_list = elems[0]['instruction_id_list']
-        # computing "pass@k" score
-        follow_instruction_list = elems[0]['follow_instruction_list']
-        for elem in elems:
-            follow_instruction_list = [
-                follow_instruction_list[i] or elem['follow_instruction_list'][i]
-                for i in range(len(follow_instruction_list))
-            ]
-
-        stats_dict['prompt']['total'] += 1
-        if all(follow_instruction_list):
-            stats_dict['prompt']['correct'] += 1
-
-        stats_dict['instruction']['total'] += len(instruction_id_list)
-        stats_dict['instruction']['correct'] += sum(follow_instruction_list)
-
-        for instruction_id, followed_or_not in zip(instruction_id_list, follow_instruction_list):
-            instruction_id = instruction_id.split(":")[0]
-            stats_dict['tier0']['total'][instruction_id] += 1
-            if followed_or_not:
-                stats_dict['tier0']['correct'][instruction_id] += 1
+    def _get_score_dict(self, prediction: dict) -> dict[bool | int | float]:
```

Comment:
```suggestion
    def _get_score_dict(self, prediction: dict) -> dict[str, bool | int | float]:
```

Reply:
str is the key is what this suggestion is trying to indicate.

Reply:
oh, right, yeah, let me change that